### PR TITLE
Ingesting Spectra for Manj20

### PR DESCRIPTION
**Pull request by SherelynA**

### CODE REVIEW

**Comment by SherelynA** [comment]
Link to Google Sheet: https://docs.google.com/spreadsheets/d/1_RxjvPuPcNVswinf1q2zzM61plaP3Z9_b7UTRJCtFxk/edit?usp=sharing

File: data/2mass_j11074656-7615174.json [hunk 1]
```diff
@@ -0,0 +1,37 @@
+{
+    "Sources": [
+        {
+            "source": "2MASS J11074656-7615174",
+            "ra": 166.944,
+            "dec": -76.2548,
+            "epoch": 2000.0,
+            "equinox": "J2000",
+            "shortname": null,
+            "reference": "G\u00f3me01",
```
Comment (kelle) [inline data/2mass_j11074656-7615174.json:10]: needs to be fixed

File: data/2mass_j11085497–7632410.json [hunk 2]
```diff
@@ -0,0 +1,37 @@
+{
+    "Sources": [
+        {
+            "source": "2MASS J11085497\u20137632410",
```
Comment (kelle) [inline data/2mass_j11085497–7632410.json:4]: \u needs to be fixed

File: data/2mass_j11085497–7632410.json [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+            "equinox": "J2000",
+            "shortname": null,
+            "reference": "Pers00",
+            "other_references": null,
+            "comments": null
+        }
+    ],
+    "Names": [
+        {
+            "other_name": "2MASS J11085497\u20137632410"
```
Comment (kelle) [inline data/2mass_j11085497–7632410.json:17]: fix \u

File: data/2mass_j11123099–7653342.json [hunk 4]
```diff
@@ -0,0 +1,37 @@
+{
+    "Sources": [
+        {
+            "source": "2MASS J11123099\u20137653342",
```
Comment (kelle) [inline data/2mass_j11123099–7653342.json:4]: fix \u

File: data/2mass_j11123099–7653342.json [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+            "ra": 168.129,
+            "dec": -76.8928,
+            "epoch": 2000.0,
+            "equinox": "J2000",
+            "shortname": null,
+            "reference": "L\u00f3pe04",
```
Comment (kelle) [inline data/2mass_j11123099–7653342.json:10]: fix

File: data/2mass_j11123099–7653342.json [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+            "other_references": null,
+            "comments": null
+        }
+    ],
+    "Names": [
+        {
+            "other_name": "2MASS J11123099\u20137653342"
```
Comment (kelle) [inline data/2mass_j11123099–7653342.json:17]: fix \u

File: data/2mass_j15591135–2338002.json [hunk 7]
```diff
@@ -0,0 +1,37 @@
+{
+    "Sources": [
+        {
+            "source": "2MASS J15591135\u20132338002",
```
Comment (kelle) [inline data/2mass_j15591135–2338002.json:4]: fix \u

File: data/2mass_j15591135–2338002.json [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+            "equinox": "J2000",
+            "shortname": null,
+            "reference": "Ardi00",
+            "other_references": null,
+            "comments": null
+        }
+    ],
+    "Names": [
+        {
+            "other_name": "2MASS J15591135\u20132338002"
```
Comment (kelle) [inline data/2mass_j15591135–2338002.json:17]: fix \u

File: data/2mass_j16060391–2056443.json [hunk 9]
```diff
@@ -0,0 +1,37 @@
+{
+    "Sources": [
+        {
+            "source": "2MASS J16060391\u20132056443",
```
Comment (kelle) [inline data/2mass_j16060391–2056443.json:4]: fix \u

File: data/2mass_j16060391–2056443.json [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+            "equinox": "J2000",
+            "shortname": null,
+            "reference": "Mart04.226",
+            "other_references": null,
+            "comments": null
+        }
+    ],
+    "Names": [
+        {
+            "other_name": "2MASS J16060391\u20132056443"
```
Comment (kelle) [inline data/2mass_j16060391–2056443.json:17]: fix \u

File: data/Publications.json [hunk 11]
```diff
@@ -6790,5 +6784,95 @@
...
+        "description": "Spectroscopic Identification of DENIS-selected Brown Dwarf Candidates in the Upper Scorpius OB Association"
+    },
+    {
+        "publication": "Pers00",
+        "bibcode": "2000A&A...357..219P",
+        "doi": null,
+        "description": "ISOCAM observations of the Chamaeleon I dark cloud"
+    },
+    {
+        "publication": "L\u00f3pe04",
```
Comment (kelle) [inline data/Publications.json:6855]: fix

File: data/Publications.json [hunk 12]
```diff
@@ -6790,5 +6784,95 @@
...
+        "bibcode": "2004A&A...416..555L",
+        "doi": "10.1051/0004-6361:20031720",
+        "description": "The brown dwarf population in the Chamaeleon I cloud"
+    },
+    {
+        "publication": "G\u00f3me01",
```
Comment (kelle) [inline data/Publications.json:6861]: fix

File: data/usco_j160606.29-233513.3.json [hunk 13]
```diff
@@ -13,6 +13,9 @@
         }
     ],
     "Names": [
+        {
+            "other_name": "2MASS J16060629\u20132335133"
```
Comment (kelle) [inline data/usco_j160606.29-233513.3.json:17]: fix

File: scripts/ingests/ingest_utils.py [hunk 14]
```diff
@@ -872,8 +872,15 @@ def ingest_spectra(db, sources, spectra, regimes, telescopes, instruments, modes
                     raise SimpleError(msg)
                 else:
                     continue
-
-            # check telescope, instrument, mode exists
+            elif db.query(db.Publications).filter(db.Publications.c.name == references[i]).count() == 0:
```
Comment (kelle) [inline scripts/ingests/ingest_utils.py:875]:
```suggestion
            if db.query(db.Publications).filter(db.Publications.c.name == references[i]).count() == 0:
```

Comment (kelle) [reply scripts/ingests/ingest_utils.py:875]: Also get rid of "TODO" on line 784

**Review by kelle — CHANGES_REQUESTED** [review]

File: data/2mass_j15591135–2338002.json (hunk 7 above)
Comment (kelle) [reply data/2mass_j15591135–2338002.json:4]: Looks like this one was missed by the most recent commit.

File: scripts/ingests/ingest_utils.py [hunk 15]
```diff
@@ -872,8 +872,15 @@ def ingest_spectra(db, sources, spectra, regimes, telescopes, instruments, modes
                     raise SimpleError(msg)
                 else:
                     continue
-
-            # check telescope, instrument, mode exists
+            if db.query(db.Publications).filter(db.Publications.c.name == references[i]).count() == 0:
```
Comment (kelle) [inline scripts/ingests/ingest_utils.py:875]: This function looks great!  Delete line 866 that contains the TODO item.

**Review by kelle — CHANGES_REQUESTED** [review]
two teeny things left -- so close!

**Review by kelle — APPROVED** [review]

**Pull request merged**